src/errors.rs

```diff
@@ -1,67 +1,211 @@
-use output;
-use std::ffi::OsString;
+use std::ffi;
+use std::fmt;
+use std::io;
 
-const ERROR_PREFIX: &'static str = "CLI assertion failed";
+use failure;
 
-fn format_cmd(cmd: &[OsString]) -> String {
+fn format_cmd(cmd: &[ffi::OsString]) -> String {
     let result: Vec<String> = cmd.iter()
         .map(|s| s.to_string_lossy().into_owned())
         .collect();
     result.join(" ")
 }
 
-error_chain! {
-    links {
-        Output(output::Error, output::ErrorKind);
+pub trait ChainFail {
+    fn chain<E>(self, cause: E) -> Self
+    where
+        E: Into<failure::Error>;
+}
+
+pub trait ResultChainExt<T> {
+    fn chain<C>(self, chainable: C) -> Result<T, C>
```

Comment:
More help for chaining.

I don't like the naming.  It is unclear in `ChainFail` and `ResultChainExt` which type is chaining to which.  I don't have better names and these are internal APIs, so I decided to not worry about it atm